memory-e2e/billing.py

```
@@ -0,0 +1,29 @@
+import hashlib
+import sqlite3
+
+import requests
+
+
+def lookup_invoice(conn: sqlite3.Connection, invoice_id: str):
+    cur = conn.cursor()
+    cur.execute("SELECT * FROM invoices WHERE id = '" + invoice_id + "'")
```

Ruff (0.16.0)

[error] Possible SQL injection vector through string-based query construction

(S608)